src/test/java/com/ksm/wstbackoffice/controller/WarehouseControllerIT.java

```diff
@@ -0,0 +1,130 @@
+package com.ksm.wstbackoffice.controller;
+
+import com.ksm.wstbackoffice.dto.AddressDto;
+import com.ksm.wstbackoffice.dto.WarehouseDto;
+import com.ksm.wstbackoffice.entity.AddressEntity;
+import com.ksm.wstbackoffice.enumeration.PriceType;
+import com.ksm.wstbackoffice.enumeration.WarehouseType;
+import com.ksm.wstbackoffice.mapper.AddressMapper;
+import com.ksm.wstbackoffice.service.AddressService;
+import com.ksm.wstbackoffice.service.IWarehouseService;
+import io.restassured.http.ContentType;
+import org.hamcrest.CoreMatchers;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlGroup;
+
+import java.util.stream.Stream;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.when;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+
+@SqlGroup({
+        @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = "classpath:sql/warehouseInsert.sql"),
+        @Sql(executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD, scripts = "classpath:sql/warehouseDelete.sql")
+})
+public class WarehouseControllerIT extends BaseControllerIT {
+    @Autowired
+    private IWarehouseService warehouseService;
+    @Autowired
+    private AddressService addressService;
+    @Autowired
+    private AddressMapper addressMapper;
+
+
+    public static Stream<Arguments> findByIdTestArguments() {
+        return Stream.of(
+                Arguments.of(-1L, 200),
+                Arguments.of(-100L, 404),
+                Arguments.of(0L, 404),
+                Arguments.of(Long.MAX_VALUE, 404)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("findByIdTestArguments")
+    public void findByIdTest(Long id, int expectedStatus) {
+        given().
+            pathParam("id", id).
+        when().
+            get("warehouses/{id}").
+        then().
+            statusCode(expectedStatus);
+    }
+
+    @Test
+    public void findAllTest() {
+        when().
+            get("warehouses").
+        then().
+            statusCode(200).
+        body("size()", CoreMatchers.equalTo(4));
```

Comment:
Can you use static import in a case like this for a CoreMatchers.equalTo?

Reply:
fixed